### PR TITLE
Solr integration

### PR DESCRIPTION
**Pull request by buchi**
Integrates ftw.solr 2.0 into GEVER

First the custom date range query handling in advanced search is removed because Plone already has it's own implementation and we don't want to support both in Solr.

Solr is used in the search form (@@search), the livesearch (@@livesearch_reply) and in catalog based tabbedview tabs if the user enters text in the filter field.

The upgrade step only installs and configures ftw.solr. It is **not** enabled by default. Enabling and indexing all content in Solr is done in an opengever.maintenance script (https://github.com/4teamwork/opengever.maintenance/blob/activate-solr/opengever/maintenance/scripts/activate_solr.py). If Solr is not enabled, GEVER still uses the catalog for searching.

The Solr server configuration containing the schema is included in this package as it depends on the GEVER release: https://github.com/4teamwork/opengever.core/tree/solr-integration/solr-conf

The installation of the Solr server is done in Buildout using the recipe ftw.recipe.solr.

### CODE REVIEW

File: opengever/core/profiles/default/registry.xml [hunk 1]
```diff
@@ -191,6 +191,15 @@
   <!-- SHARING -->
   <records interface="opengever.sharing.interfaces.ISharingConfiguration" />
 
+  <!-- Solr settings -->
+  <records interface="ftw.solr.interfaces.ISolrSettings">
+    <value key="local_query_parameters">{!boost b=recip(ms(NOW,modified),3.858e-10,10,1)}</value>
+    <value key="simple_search_term_pattern">Title:{term}^100 OR Title:{term}*^10 OR SearchableText:{term}^10 OR SearchableText:{term}* OR sequence_number:{term}^2000</value>
```
Comment (lukasgraf) [inline opengever/core/profiles/default/registry.xml]: This gave me `Invalid Number: amazonas for field sequence_number` when searching for `amazonas`. The produced query was
```
{u'filter': [u'allowedRolesAndUsers:(Manager OR Authenticated OR Anonymous OR user\\:zopemaster)'], u'query': u'{!boost b=recip(ms(NOW,modified),3.858e-10,10,1)}Title:amazonas^100 OR Title:amazonas*^10 OR SearchableText:amazonas^10 OR SearchableText:amazonas* OR sequence_number:amazonas^2000', u'params': {'fl': ['UID', 'Title', 'getIcon', 'portal_type', 'path']}, u'limit': 10, u'offset': 0}
```
(printed [at `search.py:37`](https://github.com/4teamwork/ftw.solr/blob/master/ftw/solr/search.py#L37)).

Went on with my testing by removing `OR sequence_number:{term}^2000` from `simple_search_term_pattern`, then it worked.

File: opengever/base/browser/livesearch.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+from Products.CMFCore.utils import getToolByName
+from Products.CMFPlone import PloneMessageFactory as pmf
+from Products.CMFPlone.utils import normalizeString
+from Products.CMFPlone.utils import safe_unicode
+from Products.Five.browser import BrowserView
+from Products.PythonScripts.standard import html_quote
+from zope.component import getUtility
+
+
+USE_ICON = True
```
Comment (lukasgraf) [inline opengever/base/browser/livesearch.py]: From what I can tell, Mimetype icons (for documents with files) seem to be missing from livesearch results. The old rendering including the icons can be triggered by reverting `ISearchSettings.use_solr` back to `False`.

File: opengever/tabbedview/catalog_source.py [hunk 3]
```diff
@@ -12,3 +18,57 @@
...
+            registry = getUtility(IRegistry)
+            settings = registry.forInterface(ISearchSettings)
+            if settings.use_solr:
+                return self.solr_results(query)
+
+        return super(GeverCatalogTableSource, self).search_results(query)
+
+    def solr_results(self, query):
+        solr_query = u'Title:{term} OR SearchableText:{term}'.format(
+            term=query['SearchableText'])
```
Comment (lukasgraf) [inline opengever/tabbedview/catalog_source.py]: Filtering in a tabbed view documents tab for `bücher` gives me this `UnicodeDecodeError`:

```
2018-02-08 22:29:07 ERROR Zope.SiteErrorLog 1518125347.490.499217313579 http://localhost:8080/fd/ordnungssystem/fuehrung/vertraege-und-vereinbarungen/dossier-27/@@tabbed_view/listing
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module ftw.tabbedview.browser.tabbed, line 167, in listing
  Module opengever.tabbedview.browser.tabs, line 58, in __call__
  Module opengever.tabbedview.browser.tabs, line 61, in render_preferred_view
  Module opengever.tabbedview.browser.tabs, line 185, in __call__
  Module ftw.tabbedview.browser.listing, line 109, in __call__
  Module ftw.tabbedview.browser.listing, line 197, in update
  Module opengever.tabbedview.catalog_source, line 29, in search_results
  Module opengever.tabbedview.catalog_source, line 35, in solr_results
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 1: ordinal not in range(128)
```

**Comment by lukasgraf** [comment]
Minor side node: The highlighted search terms currently look a bit wonky when they happen to be part of the item's title:
snippet_title

In the description, the could be styled slightly more prominent IMHO. `</nitpick>`

**Comment by lukasgraf** [comment]
EML E-Mails don't get indexed at all, Tika in Solr 7.2 fails with `java.lang.NoClassDefFoundError: org/apache/james/mime4j/stream/MimeConfig$Builder`.

This seems to be resolved in Solr 7.3:
https://issues.apache.org/jira/browse/SOLR-11622

File: opengever/core/profiles/default/registry.xml (hunk 1 above)
Comment (buchi) [reply opengever/core/profiles/default/registry.xml]: I've remove the sequence_number from the pattern for now. Will need to find a better solution to handle this.

File: opengever/tabbedview/catalog_source.py (hunk 3 above)
Comment (buchi) [reply opengever/tabbedview/catalog_source.py]: Fixed.

**Comment by buchi** [comment]
Filtering trash items requires https://github.com/4teamwork/ftw.solr/pull/109

**Pull request merged**